src/conn.cpp

```diff
@@ -121,6 +122,25 @@ void ConnBase::bevRead()
                        "%s %s Receive header\n", peerLabel(), peerName.c_str());
 
         if(header[2]&pva_flags::Control) {
+            if(header[3]==pva_ctrl_msg::SetEndian) {
+                /* This should be the first message sent by a (supposedly) a server.
+                 * However, old pvAccess* accepts it from either peer at any time.
+                 *
+                 * The protocol spec. claims that we should inspect the size field
+                 * (bytes 4-7) and act as follows.
+                 * 0x00000000 - Send future messages using endianness on this (received)
+                 *              message.  Peer will ignore MSB flag in our headers!
+                 * 0xffffffff - Send future messages as we like.  Peer will test the
+                 *              MSB flag.
+                 *
+                 * However, neither pvAccessCPP nor pvAccessJava actually test this.
+                 * Instead the 0x00000000 behavior is assumed.
+                 *
+                 * So we latch the byte order here, as the peer will ignore the MSB
+                 * flag in our messages...
+                 */
+                sendBE = header[2]&pva_flags::MSB;
```

Comment:
The new java implementation behaves like this:
The 'size' field 0x00000000 or 0xffffffff is indeed ignored.
The MSB flag is used to set the equivalent of the `sendBE` flag, and future messages that the client *sends* will always use the endianess that the server configured via the set-endian command. For *received* messages, however, we continue to check the byte order flag in each received messages,

This specific PVXS change is only about the `sendBE`, locking the byte order of messages that the client sends to the server from now on, making 0x00000000  the default and only option. 
What about received messages? Should the set-endian command also affect received messages, we trust the server to from now on only give us messages in that same byte order, or should the client still check the byte order flag of each received message?